### ISSUE TITLE
(URGENT) site generation failure due to relaton-bib

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-bib#45]
From https://github.com/metanorma/ogc-dggs-xmi

Run:
```
bundle exec metanorma site generate
```

```
[info]: Building collection file: /Users/me/src/mn/ogc-dggs-xmi/site/documents.xml ...
[info]: Generating html site in /Users/me/src/mn/ogc-dggs-xmi/site ...
bundler: failed to load command: metanorma (/Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/bin/metanorma)
Traceback (most recent call last):
	58: from /Users/me/.rbenv/versions/2.6.6/bin/bundle:23:in `<main>'
	57: from /Users/me/.rbenv/versions/2.6.6/bin/bundle:23:in `load'
	56: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/exe/bundle:37:in `<top (required)>'
	55: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/friendly_errors.rb:130:in `with_friendly_errors'
	54: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/exe/bundle:49:in `block in <top (required)>'
	53: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli.rb:24:in `start'
	52: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	51: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli.rb:30:in `dispatch'
	50: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	49: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	48: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	47: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli.rb:494:in `exec'
	46: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli/exec.rb:28:in `run'
	45: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli/exec.rb:63:in `kernel_load'
	44: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.15/lib/bundler/cli/exec.rb:63:in `load'
	43: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/bin/metanorma:23:in `<top (required)>'
	42: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/bin/metanorma:23:in `load'
	41: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/exe/metanorma:20:in `<top (required)>'
	40: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli.rb:67:in `start'
	39: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
	38: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	37: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	36: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-hollaback-0.2.0/lib/thor/hollaback.rb:62:in `run'
	35: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	34: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor.rb:243:in `block in subcommand'
	33: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/invocation.rb:116:in `invoke'
	32: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	31: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	30: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-hollaback-0.2.0/lib/thor/hollaback.rb:62:in `run'
	29: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	28: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/commands/site.rb:25:in `generate'
	27: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/site_generator.rb:23:in `generate'
	26: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/site_generator.rb:30:in `generate'
	25: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/site_generator.rb:30:in `chdir'
	24: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/site_generator.rb:32:in `block in generate'
	23: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/metanorma-cli-1.4.7.1/lib/metanorma/cli/site_generator.rb:87:in `convert_to_html_page'
	22: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/base_convertor.rb:43:in `to_html'
	21: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/base_convertor.rb:22:in `to_html'
	20: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/base_convertor.rb:52:in `convert_to_html'
	19: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/xml_to_html_renderer.rb:35:in `render'
	18: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/xml_to_html_renderer.rb:18:in `render'
	17: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/xml_to_html_renderer.rb:49:in `build_liquid_document'
	16: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli/xml_to_html_renderer.rb:114:in `build_bibcollection'
	15: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/bibcollection.rb:41:in `from_xml'
	14: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/bibcollection.rb:41:in `map'
	13: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:237:in `each'
	12: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:237:in `upto'
	11: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:238:in `block in each'
	10: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/bibcollection.rb:44:in `block in from_xml'
	 9: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/bibdata.rb:41:in `from_xml'
	 8: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-cli-1.7.3/lib/relaton/cli.rb:64:in `parse_xml'
	 7: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:11:in `from_xml'
	 6: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:34:in `item_data'
	 5: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:220:in `fetch_dates'
	 4: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:220:in `reduce'
	 3: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:237:in `each'
	 2: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:237:in `upto'
	 1: from /Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/nokogiri-1.10.10/lib/nokogiri/xml/node_set.rb:238:in `block in each'
/Users/me/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:223:in `block in fetch_dates': undefined method `<<' for nil:NilClass (NoMethodError)
```

**Comment by ronaldtse** [comment]
The failure is seen here: https://github.com/metanorma/ogc-dggs-xmi/pull/7

**Comment by ronaldtse** [comment]
Happens on Windows GHA too, somehow not on Ubuntu and macOS on GHA:

https://github.com/metanorma/ogc-dggs-xmi/pull/7/checks?check_run_id=2378278554

```
C:/Users/RUNNER~1/AppData/Local/Temp/ocr3045.tmp/lib/ruby/gems/2.4.0/gems/relaton-bib-1.7.4/lib/relaton_bib/xml_parser.rb:224:in `block in fetch_dates': undefined method `<<' for nil:NilClass (NoMethodError)
415
	from C:/Users/RUNNER~1/AppData/Local/Temp/ocr3045.tmp/lib/ruby/gems/2.4.0/gems/nokogiri-1.10.10-x64-mingw32/lib/nokogiri/xml/node_set.rb:238:in `block in each'
416
```

**Comment by andrew2net** [comment]
@ronaldtse fixed in v 1.7.5
